croncat/src/store/agent.rs

```diff
@@ -140,11 +140,17 @@ impl LocalAgentStorage {
     }
 
     /// Generate a new account_id to the local storage.
-    pub fn generate_account(&mut self, account_id: AccountId) -> Result<(), Report> {
+    pub fn generate_account(&mut self, account_id: AccountId, mnemonic: Option<String>) -> Result<(), Report> {
         match self.get(&account_id) {
             Some(_) => Err(eyre!(r#"Agent "{account_id}" already created"#)),
             None => {
-                self.insert(account_id.clone(), Mnemonic::generate(24).unwrap())?;
+                let mnemonic2 = if let Some(phrase) = mnemonic {
+                    phrase
+                } else {
+                    Mnemonic::generate(24).unwrap().to_string()
+                };
+                let parsed = Mnemonic::parse_normalized(&mnemonic2);
```

Comment:
Nice, and it checks for an invalid word count too 
```
   mnemonic has a word count that is not a multiple of 6: 11
```

Reply:
oh neat